torchgeo/datasets/seasonet.py

```diff
@@ -0,0 +1,456 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""SeasoNet dataset."""
+
+import os
+import random
+from typing import Callable, Collection, Iterable, Optional
+
+import matplotlib.patches as mpatches
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
```

Comment:
We don't (yet) have a required dep on pandas, this should be imported in the dataset class. See gbif and other datasets for examples of how to do this and test this.